src/cloud/index.js

```diff
@@ -0,0 +1,193 @@
+
+  const config = await getConfig(environment);
+  await checkConfig(environment, config);
+
+  await rolloutDeployment(environment, service, subservice);
+}
+
+export async function deploy(options) {
+  const { environment, service, subservice, tag } = options;
+  if (!devMode) assertBedrockRoot();
+
+  const config = await getConfig(environment);
+  await checkConfig(environment, config);
+  const { project } = config.gcloud;
+  const platformName = getPlatformName();
+
+  await warn(environment);
+
+  await buildImage(platformName, service, subservice, tag);
+  await dockerPush(project, platformName, service, subservice, tag);
+  await rolloutDeployment(environment, service, subservice);
+}
+
+export async function undeploy(options) {
+  const { environment, service, subservice } = options;
+  if (!devMode) assertBedrockRoot();
+
+  const config = await getConfig(environment);
+  await checkConfig(environment, config);
+  await checkDeployment(service, subservice);
+
+  await deleteDeployment(environment, service, subservice);
+}
+
+export async function info(options) {
+  const { environment, service, subservice } = options;
+  if (!devMode) assertBedrockRoot();
+
+  const config = await getConfig(environment);
+  await checkConfig(environment, config);
+  const deployment = getDeployment(service, subservice);
+  const deploymentInfo = await checkDeployment(service, subservice);
+  const { annotations } = deploymentInfo.spec.template.metadata;
+  console.info(`Deployment "${deployment}" annotations:`);
+  console.info(annotations);
+}
+
+export async function provision(options) {
+  const { environment, terraform } = options;
+  if (!devMode) assertBedrockRoot();
+
+  const config = await getConfig(environment);
+  await checkGCloudProject(config.gcloud);
+
+  await provisionTerraform(environment, terraform, config.gcloud);
+}
+
+export async function shell(options) {
+  const { environment, service, subservice } = options;
+  if (!devMode) assertBedrockRoot();
+
+  const config = await getConfig(environment);
+  await checkConfig(environment, config);
+
+  const podsJSON = await exec(`kubectl get pods -o jsonpath='{.items}' --ignore-not-found`);
+  if (!podsJSON) {
+    console.info(kleur.yellow(`No running pods`));
+    process.exit(0);
+  }
+  const pods = JSON.parse(podsJSON.slice(1, -1));
+
+  let deployment = 'api-cli-deployment';
+  if (service) {
+    deployment = getDeployment(service, subservice);
+  }
+
+  const filteredPods = pods.filter((pod) => pod.metadata.name.startsWith(deployment));
+
+  if (!filteredPods.length) {
+    console.info(kleur.yellow(`No running pods for deployment "${deployment}"`));
+    process.exit(0);
+  }
+
+  const podName = filteredPods[0].metadata.name;
+  console.info(kleur.green(`=> Starting bash for pod: "${podName}"`));
+
+  const { spawn } = require('child_process');
+
+  const child = spawn('kubectl', ['exec', '-it', podName, '--', 'bash'], {
+    stdio: 'inherit',
```

Comment:
I'm assuming that opening a shell into a pod is a one off thing that goes beyond what `exec` can handle, so it needs to be done custom like this?

Reply:
Yes, you basically hand off TTY control to the spawend process.

Reply:
cool makes sense 👍